llvm-external-projects/iree-compiler-api/lib/CAPI/Compiler.cpp

```diff
@@ -59,7 +64,28 @@ struct CompilerOptions {
 
 DEFINE_C_API_PTR_METHODS(IreeCompilerOptions, CompilerOptions)
 
-void ireeCompilerRegisterTargetBackends() { registerHALTargetBackends(); }
+void ireeCompilerRegisterAllDialects(MlirContext context) {
+  DialectRegistry registry;
+  registerAllDialects(registry);
+  unwrap(context)->appendDialectRegistry(registry);
+}
+
+void ireeCompilerRegisterAllPasses() {
+  registerAllPasses();
+}
+
+void ireeCompilerRegisterAllIREETranslations() {
+  registerMlirTranslations();
+  registerIreeTranslations();
+}
+
+void ireeCompilerRegisterVMTargets() {
```

Comment:
Looking at this more closely, I think this one should not be needed either.

Reply:
Indeed, seems like not needed.